### PR TITLE
Add DynamoDb implementation for KVStore

### PR DESCRIPTION
**Pull request by crioux-stripe**
### Context

This is actually @mcowgill-stripe 's work but he's out of office this week and I took the liberty of altering the unit tests somewhat.

This pull request adds a DynamoDB implementation of the Key Value store which should be very useful to other potential OSS adopters who want to spin this up within their AWS accounts. I'd like to use this contribution, and a few other incoming ones as a basis for developing trust and potentially opening up contributor status to some Stripes in the medium term.

### Checklist

- [X] `./gradlew build` compiles code correctly
- [X] Added new tests where applicable
- [X] `./gradlew test` passes all tests (Passes the same number it did previously.)
- [X] Extended README or added javadocs where applicable

### CODE REVIEW

**Comment by crioux-stripe** [comment]
As a note, because I imagine it'll come up. The `mantis-control-plane/mantis-control-plane-server/dynamodb-local-metadata.json` needs to be present and empty for the tests to pass. If it isn't present it'll create one, populate it, and fail for whatever reason.

File: mantis-control-plane/mantis-control-plane-server/build.gradle
Comment (Andyz26) [inline mantis-control-plane/mantis-control-plane-server/build.gradle:39]: I feel like it's better to move this to its own module so we can avoid adding dynamodb SDK dep to the control-plane-server everywhere?

Comment (crioux-stripe) [reply mantis-control-plane/mantis-control-plane-server/build.gradle:39]: @Andyz26 I agree, let me get this moved into a new module today.

**Comment by crioux-stripe** [comment]
@Andyz26 I got it moved into a separate but parallel module. I debated top level modules vs nesting them under `mantis-control-plane` and decided on nesting due to the tight coupling -- the store is only for the control plane.

**Comment by crioux-stripe** [comment]
Hrmm. This seems to compile fine with Java 11. I'm not sure if I can get 8 installed on this work machine to debug. Will look into this and circle back.

**Comment by crioux-stripe** [comment]
All should be well now, it turns out the `2.+` versions of `DynamoDbLocal` did not have builds for Java 8 published. Downgrading to `1.25.0` fixed the issue and appears to be compatible with our limited usage.

**Comment by crioux-stripe** [comment]
Oh man. Spotless got me. Pushed a fix.

**Review by Andyz26 — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/src/main/java/io/mantisrx/server/master/store/DynamoStore.java
Comment (crioux-stripe) [reply mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/src/main/java/io/mantisrx/server/master/store/DynamoStore.java]: Good call! Pushed a commit switching to the annotation based logger instantiation.

**Comment by crioux-stripe** [comment]
Merging the base branch in again. I'm curious if there is any action left on my end or should we merge this change?

**Comment by crioux-stripe** [comment]
@Andyz26 I don't think I have merge rights!

**Comment by codyrioux** [comment]
Checking on this account. I also don’t have merge rights here. I’ll need someone from Netflix to take this across the finish line — unless you’re willing to pull me back into the fold. :)

File: mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/build.gradle
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/build.gradle:58]: do we need these definitions?

Comment (sundargates) [reply mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/build.gradle:58]: Also this file mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/dynamodb-local-metadata.json looks empty.

**Comment by sundargates** [comment]
> Checking on this account. I also don’t have merge rights here. I’ll need someone from Netflix to take this across the finish line — unless you’re willing to pull me back into the fold. :)

I'll merge this change after the build finishes.

**Review by sundargates — APPROVED** [review]
THank you for this change!

File: mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/build.gradle
Comment (crioux-stripe) [reply mantis-control-plane/mantis-control-plane-store/mantis-control-plane-store-dynamodb/build.gradle:58]: @sundargates The `dynamodb-local-metadata.json` file needs to exist and be empty, otherwise it creates one and populates it with some data and the tests fail.

**Pull request merged**
